packages/drivers/odsp-driver/src/buildOdspShareLinkReqParams.ts

```diff
@@ -0,0 +1,18 @@
+import { ShareLinkTypes, ShareLink } from "@fluidframework/odsp-driver-definitions";
+
+export function buildOdspShareLinkReqParams(shareLinkType: ShareLinkTypes | ShareLink | undefined) {
+    let shareLinkRequestParams = "";
+    if (shareLinkType) {
+        const linkScope = (shareLinkType as ShareLink).linkScope;
+        if (linkScope) {
+            shareLinkRequestParams = `createLinkScope=${linkScope}`;
+            const linkRole = (shareLinkType as ShareLink).linkRole;
+            if (linkRole) {
+                shareLinkRequestParams += `&createLinkRole=${linkRole}`;
+            }
+        } else {
+            shareLinkRequestParams = `createLinkType=${shareLinkType}`;
+        }
+    }
+    return shareLinkRequestParams;
+}
```

Comment:
I'd rewrite this using guard clauses https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

I think it would make it easier to read and reason about by reducing nesting

```
if (shareLinkType === undefined) {
    return ""
}

const linkScope = (shareLinkType as ShareLink).linkScope;
if (linkScope === undefined) {
    return `createLinkType=${shareLinkType}`;
}

etc..
```

Reply:
done